File: June 2025/14.cpp
#include <iostream>
using namespace std;

/*
class Node {
public:
    int data;
    Node *left, *right;

    Node(int val) {
        data = val;
        left = right = nullptr;
    }
};
*/

class Solution {
  public:
    // Helper function to check if two subtrees are mirror images
    bool isMirror(Node* left, Node* right) {
        if (!left && !right) return true;
        if (!left || !right) return false;
        return (left->data == right->data) &&
               isMirror(left->left, right->right) &&
               isMirror(left->right, right->left);
    }

    // Main function to check symmetry
    bool isSymmetric(Node* root) {
        if (!root) return true;
        return isMirror(root->left, root->right);
    }
};

File: June 2025/14.py
class Node:
    def __init__(self, val):
        self.right = None
        self.data = val
        self.left = None

class Solution:
    def isSymmetric(self, root):
        # Helper function to check if two subtrees are mirror images
        def isMirror(left, right):
            if not left and not right:
                return True
            if not left or not right:
                return False
            return (left.data == right.data) and isMirror(left.left, right.right) and isMirror(left.right, right.left)
        
        # If root is None, it's symmetric
        if not root:
            return True
        return isMirror(root.left, root.right)
